### ISSUE TITLE
OpenGL apps are using software rendering when using udev backend in anvil

### ISSUE THREAD

**Issue by kchibisov** [Smithay/smithay#187]
When I run alacritty in anvil with `--tty-udev` it's using software rendering via llvmpipe, `--winit` works fine. When I run `list_globals` example of `wayland-client` I get:
with `--tty-udev`
```
1: wl_shm (version 1)
2: wl_compositor (version 4)
3: wl_subcompositor (version 1)
4: xdg_wm_base (version 1)
5: zxdg_shell_v6 (version 1)
6: wl_shell (version 1)
7: wl_data_device_manager (version 3)
8: wl_seat (version 5)
9: wl_output (version 3)
```

and with `--winit`
```
1: wl_drm (version 2)
2: wl_shm (version 1)
3: wl_compositor (version 4)
4: wl_subcompositor (version 1)
5: xdg_wm_base (version 1)
6: zxdg_shell_v6 (version 1)
7: wl_shell (version 1)
8: wl_data_device_manager (version 3)
9: wl_seat (version 5)
10: wl_output (version 3)
```

anvil was with default features on systemd based system.

**Comment by vberger** [comment]
Probably related to the fact reported in the chatroom that `smithay::backend::udev::primary_gpu` is apparently broken.

Anvil only initializes the EGL support on the primary GPU, which it fails to find given this function is broken.

**Comment by kchibisov** [comment]
Was fixed by 39500f1d052a92077d95e648eda70aff66c31ace
